flecsi/exec/hpx/policy.cc

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2016, Triad National Security, LLC
+// All rights reserved.
+
+#include <flecsi-config.h>
+
+#if !defined(FLECSI_ENABLE_HPX)
+#error FLECSI_ENABLE_HPX not defined! This file depends on HPX!
+#endif
+
+#include <hpx/modules/execution_base.hpp> // yield_while
+#include <hpx/modules/runtime_local.hpp> // get_thread_manager
+
+#include "flecsi/exec/backend.hh"
+#include "flecsi/exec/hpx/policy.hh"
+
+#include <cstdint>
+
+namespace flecsi::exec::detail {
+
+// wait for all current HPX tasks to finish executing
+void
+termination_detection() {
+  auto & tm = hpx::threads::get_thread_manager();
+  hpx::util::yield_while(
```

Comment:
Does this need to wait to avoid spinning?

Reply:
This yields the current thread to allow for the scheduler to run other possibly still pending tasks in case that the application runs on one core only. In that sense, it's not really 'spinning'. The current thread will simply be resubmitted to the end of the current task-queue, so the probability of it retrying the condition is low. The best of all solutions would be to have this functionality exposed by HPX as an API function (it is essentially relying on HPX internals here), but alas - this would require yet another HPX release...

Reply:
OK.  I was imagining that this code would run in parallel with the tasks on which it was waiting, but if it's cooperatively scheduled there isn't any problem.